pages/cql/ccs-logic-2018.cql

```diff
@@ -0,0 +1,163 @@
+*/
+
+parameter "Measurement Period" Interval<DateTime>
+
+define "Lookback Interval Two More Years":
+	Interval[start of "Measurement Period" - 2 years, end of "Measurement Period")
+
+define "Lookback Interval Four More Years":
+	Interval[start of "Measurement Period" - 4 years, end of "Measurement Period")
+
+/*
+This library evaluates with respect to exactly 1 candidate patient at a time,
+that patient being given by the special context parameter Patient.
+*/
+
+context Patient
+
+/*
+Eligible Population
+Product lines -- Commercial, Medicaid (report each product line separately).
+*/
+
+define "Is In Eligible Population":
+	"Is Female"
+		and "Is Age 24 to 64 at End"
+		and (not "Is In Hospice")
+
+define "Is Female":
+	Patient.gender.value = 'female'
+
+define "Is Age 24 to 64 at End":
+	AgeInYearsAt(end of "Measurement Period") between 24 and 64
+
+define "Is In Hospice":
+	false
+	// TODO: Determine properly whether the patient was in hospice during the measurement year.
+
+define "Is In Eligible Population Commercial":
+	"Is In Eligible Population"
+		and "Is Continuous Enrollment Commercial"
+
+define "Is Continuous Enrollment Commercial":
+	true
+	// TODO: Determine continuous enrollment over the 3 year period consisting
+	// of the measurement year plus the 2 prior years.
+	// Allowable gap -- No more than one gap in enrollment of up to 45 days during
+	// each year of continuous enrollment.
+	// See http://hl7.org/fhir/enrollmentresponse.html which is currently
+	// marked as a stub / draft / incomplete.
+
+define "Is In Eligible Population Medicaid":
+	"Is In Eligible Population"
+		and "Is Continuous Enrollment Medicaid"
+
+define "Is Continuous Enrollment Medicaid":
+	true
+	// TODO: Determine continuous enrollment over the measurement year.
+	// Allowable gap -- No more than one gap in enrollment of up to 45 days during
+	// each year of continuous enrollment. To determine continuous enrollment
+	// for a Medicaid beneficiary for whom enrollment is verified monthly, the
+	// member may not have more than a 1-month gap in coverage (i.e., a member
+	// whose coverage lapses for 2 months [60 days] is not considered continuously enrolled).
+	// See http://hl7.org/fhir/enrollmentresponse.html which is currently
+	// marked as a stub / draft / incomplete.
+
+/*
+Administrative Specification
+*/
+
+define "Is In Denominator":
+	"Is In Eligible Population"
+
+define "Is In Numerator":
+	case
+		when (not "Is In Eligible Population") then false
+		when "Is Cervical Cytology Test In Last 3 Years" then true
+		when (not "Is Age 30 to 64 at End") then false
+		when "Is Cervical Cytology Plus HPV Test In Last 5 Years" then true
+		else false
+	end
+
+define "Is Age 30 to 64 at End":
+	AgeInYearsAt(end of "Measurement Period") between 30 and 64
+
+define "Is Cervical Cytology Test In Last 3 Years":
+	exists(
+		"Dates of Cervical Cytology Tests" WhenCC
+			where (WhenCC as dateTime).value during "Lookback Interval Two More Years"
```

Comment:
Also, I believe if WhenCC is a Period and you cast it to dateTime (without specifying the start or end of the Period), the returned value would be an empty list.